pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/SegmentFetcherFactory.java

```diff
@@ -15,44 +15,53 @@
  */
 package com.linkedin.pinot.common.segment.fetcher;
 
-import java.util.Arrays;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
-
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.linkedin.pinot.common.utils.CommonConstants;
 import org.apache.commons.configuration.Configuration;
 import org.apache.commons.configuration.ConfigurationMap;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.linkedin.pinot.common.utils.CommonConstants;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class SegmentFetcherFactory {
   private static final Logger LOGGER = LoggerFactory.getLogger(SegmentFetcherFactory.class);
 
   public static String SEGMENT_FETCHER_CLASS_KEY = "class";
-  public static String SEGMENT_FETCHER_PROTOCOL_KEY = "protocol";
 
-  private static Map<String, SegmentFetcher> SEGMENT_FETCHER_MAP = new ConcurrentHashMap<String, SegmentFetcher>();
+  private static Map<String, SegmentFetcher> SEGMENT_FETCHER_MAP = new ConcurrentHashMap<>();
 
   static {
     SEGMENT_FETCHER_MAP.put("file", new LocalFileSegmentFetcher());
     SEGMENT_FETCHER_MAP.put("http", new HttpSegmentFetcher());
     SEGMENT_FETCHER_MAP.put("https", new HttpSegmentFetcher());
+    SEGMENT_FETCHER_MAP.put("hdfs", new HdfsSegmentFetcher());
   }
 
   public static void initSegmentFetcherFactory(Configuration pinotHelixProperties) {
     Configuration segmentFetcherFactoryConfig =
         pinotHelixProperties.subset(CommonConstants.Server.PREFIX_OF_CONFIG_OF_SEGMENT_FETCHER_FACTORY);
 
+    // initialize predefined fetcher
+    for (String protocol: SEGMENT_FETCHER_MAP.keySet()) {
+      LOGGER.debug("initializing segment fetcher protocol {}", protocol);
```

Comment:
this can be info since its one time during start up

Reply:
updated all start up log to info